### ISSUE TITLE
Getting JDBC to retry upon startup

### ISSUE THREAD

**Issue by aclowkey** [jooby-project/jooby#1218]
If the JDBC connection won't work on startup, Jooby will hang forever.

I've encountered the issue when deploying a Kubernetes cluster,  my Jooby instance starts faster than MySQL and gets communication link failure and just hangs. As soon as MySQL start it keeps running, while Jooby is hanging 😴

Is there a way to configure either Jooby or HikariCP to try to reconnect periodically?

**Comment by jknack** [comment]
This is HikariCP question, but what is the exception? 

Seems HikariCP handles connection retries https://github.com/brettwooldridge/HikariCP/issues/907

Also, have a look at https://github.com/brettwooldridge/HikariCP/wiki/Bad-Behavior:-Handling-Database-Down

**Comment by aclowkey** [comment]
I found the reason, initializationFailTimeout is by default set to 1, so if a connection cannot be made the pool will fail fast. Setting it to -1 will bypass that, and will try to get connection in background.

I've tried it as follows
1. Shutdown MySQL
2. Start jooby (Start successfully
3. Query my database dependent endpoint - fails on connection refused
4. Start MySQL
5. Query the endpoint again. Now it works!

Quote from HikariCP documentation:

> A value less than zero will bypass any initial connection attempt, and the pool will start immediately while trying to obtain connections in the background. Consequently, later efforts to obtain a connection may fail. Default: 1
